ngraph/core/src/pass/make_stateful.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/pass/make_stateful.hpp"
+
+#include <memory>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/op/util/variable.hpp>
+#include <openvino/opsets/opset8.hpp>
+
+using namespace std;
+using namespace ngraph;
+using namespace opset8;
+using namespace op::util;
+
+namespace {
+string generate_variable_name(const shared_ptr<Parameter>& param, const shared_ptr<Result>& res, uint64_t idx) {
+    return param->get_friendly_name() + res->get_friendly_name() + std::to_string(idx);
```

Comment:
I'm wondering whether an index is required here?

Reply:
we agreed to add idx to avoid duplicate names
but I'm not sure it's really required here.

Reply:
@itikhono Does it mean that users can have different names on the same variables depends on the command line? It is not a problem currently, I'm just wondering whether we can confuse a user by different names or not. For example, user creates an application for OV taking into account that input-output pairs will become states with some fixed names and doing something with these variables into the code. And after sequential mo convert with another command-line application will stop working.

Reply:
Ok, I agree. I will remove idx

Reply:
fixed